The patch is imported & adapted from an issue from upstream. A bug of build file in angle may prevent chromium from building. This patch is to fix the bug.

Also appeared in repository of Arch Linux.

Reference: 
1. https://bugs.chromium.org/p/angleproject/issues/detail?id=7582
2. https://github.com/archlinux/svntogit-packages/blob/d3575029b752eca7af94787c03224dc4719e85d1/trunk/angle-wayland-include-protocol.patch

Signed-off-by: Camber Huang <camber@aosc.io>
---

diff -upr third_party/angle.orig/BUILD.gn third_party/angle/BUILD.gn
--- a/third_party/angle.orig/BUILD.gn	2022-08-17 19:38:11.000000000 +0000
+++ b/third_party/angle/BUILD.gn	2022-08-18 11:04:09.061751111 +0000
@@ -489,6 +489,12 @@ config("angle_vulkan_wayland_config") {
   if (angle_enable_vulkan && angle_use_wayland &&
       defined(vulkan_wayland_include_dirs)) {
     include_dirs = vulkan_wayland_include_dirs
+  } else if (angle_enable_vulkan && angle_use_wayland) {
+    include_dirs = [
+      "$wayland_gn_dir/src/src",
+      "$wayland_gn_dir/include/src",
+      "$wayland_gn_dir/include/protocol",
+    ]
   }
 }
 
@@ -1073,6 +1079,7 @@ if (angle_use_wayland) {
     include_dirs = [
       "$wayland_dir/egl",
       "$wayland_dir/src",
+      "$wayland_gn_dir/include/protocol",
     ]
   }
 
diff -upr third_party/angle.orig/src/third_party/volk/BUILD.gn third_party/angle/src/third_party/volk/BUILD.gn
--- a/third_party/angle.orig/src/third_party/volk/BUILD.gn	2022-08-17 19:38:12.000000000 +0000
+++ b/third_party/angle/src/third_party/volk/BUILD.gn	2022-08-18 11:04:36.499828006 +0000
@@ -21,6 +21,9 @@ source_set("volk") {
   configs += [ "$angle_root:angle_no_cfi_icall" ]
   public_deps = [ "$angle_vulkan_headers_dir:vulkan_headers" ]
   if (angle_use_wayland) {
-    include_dirs = [ "$wayland_dir/src" ]
+    include_dirs = [
+      "$wayland_dir/src",
+      "$wayland_gn_dir/include/protocol",
+    ]
   }
 }
